docs/user-guide/tutorials/k8s-metrics-monitor.md

```diff
@@ -0,0 +1,247 @@
+It provides visibility into the health, performance, and resource utilization of your containerized applications and infrastructure.
+
+Key aspects of Kubernetes monitoring include:
+
+- **Resource Metrics**: CPU, memory, disk, and network usage for nodes, pods, and containers
+- **Cluster Health**: Status of cluster components like kube-apiserver, etcd, and controller-manager
+- **Application Metrics**: Custom metrics from your applications running in the cluster
+- **Events and Logs**: Kubernetes events and container logs for troubleshooting
+
+Effective monitoring helps you:
+- Detect and diagnose issues before they impact users
+- Optimize resource utilization and reduce costs
+- Plan capacity based on historical trends
+- Ensure SLA compliance
+- Troubleshoot performance bottlenecks
+
+## Architecture Overview
+
+The monitoring architecture consists of the following components:
+
+![Kubernetes Monitoring Architecture](/k8s-metrics-monitor-architecture.drawio.svg)
+
+**Components:**
+
+- **kube-state-metrics**: Exports cluster-level metrics about Kubernetes objects (deployments, pods, services, etc.)
+- **Node Exporter**: Exports hardware and OS-level metrics from each Kubernetes node
+- **Prometheus Operator**: Automates Prometheus deployment and configuration using Kubernetes custom resources
+- **GreptimeDB**: Acts as the long-term storage backend for Prometheus metrics with high compression and query performance
+- **Grafana**: Provides dashboards and visualizations for metrics stored in GreptimeDB
+
+## Prerequisites
+
+Before starting, ensure you have:
+
+- A running Kubernetes cluster (version >= 1.18)
+- `kubectl` configured to access your cluster
+- [Helm](https://helm.sh/docs/intro/install/) v3.0.0 or higher installed
+- Sufficient cluster resources (at least 2 CPU cores and 4GB memory available)
+
+## Install GreptimeDB
+
+GreptimeDB serves as the long-term storage backend for Prometheus metrics.
+For detailed installation steps,
+please refer to the [Deploy GreptimeDB Cluster](/user-guide/deployments-administration/deploy-on-kubernetes/deploy-greptimedb-cluster.md) documentation.
+
+### Verify the GreptimeDB Installation
+
+After deploying GreptimeDB, verify that the cluster is running.
+In this guide we assume the GreptimeDB cluster is deployed in the `greptime-cluster` namespace and named `greptimedb`.
+
+```bash
+kubectl -n greptime-cluster get greptimedbclusters.greptime.io greptimedb
+```
+
+```bash
+NAME         FRONTEND   DATANODE   META   FLOWNODE   PHASE     VERSION   AGE
+greptimedb   1          1          1      1          Running   v0.17.2   33s
+```
+
+Check the pods:
+
+```bash
+kubectl get pods -n greptime-cluster
+```
+
+```bash
+greptimedb-datanode-0                 1/1     Running   0          44s
+greptimedb-flownode-0                 1/1     Running   0          28s
+greptimedb-frontend-8bf9f558c-7wdmk   1/1     Running   0          34s
+greptimedb-meta-fc4ddb78b-nv944       1/1     Running   0          50s
```

Comment:
Use a cluster with at least 2 datanodes. In this guide, we need to cover how to use partitions to make full usage of all datanodes.